cfme/tests/containers/test_tags.py

```diff
@@ -44,37 +50,67 @@ def get_collection_entity(appliance, collection_name, provider):
         return item_collection.instantiate(**d)
 
 
+def verify_tags(obj_under_test, tag, details, dashboard, soft_assert):
+
+    obj_under_test.add_tag(tag=tag, details=details, dashboard=dashboard)
+
+    tags = obj_under_test.get_tags()
+
+    assert any(
+        object_tags.category.display_name == tag.category.display_name and
+        object_tags.display_name == tag.display_name for object_tags in tags), (
+        "{}: {} not in ({})".format(tag.category.display_name, tag.display_name, str(tags)))
+
+    obj_under_test.remove_tag(tag=tag, details=details)
+
+    post_remove_tags = obj_under_test.get_tags()
+    if post_remove_tags:
+        for post_tags in post_remove_tags:
+            soft_assert(
```

Comment:
Q:
why `soft_assert`? As per tag behavior if this condition not satisfy then test should fails. Please, correct me about this behavior.